### PR TITLE
stdlib v9 support, use namespaced stdlib functions

### PR DESCRIPTION
**Pull request by ggabijaa**
#### Pull Request (PR) description
Added Stdlib v9 support
Added Puppet v8 support

#### This Pull Request (PR) fixes the following issues
Fixes https://github.com/voxpupuli/puppet-elasticsearch/issues/1201

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     },
     {
       "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 4.13.0 < 9.0.0"
+      "version_requirement": ">= 4.13.0 < 10.0.0"
```
Comment (smortex) [inline metadata.json:25]: The namespaced version of merge() was added in [04d1f0c5cccb93d0bc928c7623c4b0c0ebda5b6b](https://github.com/puppetlabs/puppetlabs-stdlib/commit/04d1f0c5cccb93d0bc928c7623c4b0c0ebda5b6b) which require version 9.0.0 of the stdlib module:

```suggestion
      "version_requirement": ">= 9.0.0 < 10.0.0"
```

**Review by smortex — COMMENTED** [review]
CI failures seems unrelated.  Maybe a modulesync update will help in this regard.

File: metadata.json (hunk 1 above)
Comment (ggabijaa) [reply metadata.json]: updated accordingly, thanks

**Comment by ldaneliukas** [comment]
@smortex any updates on this PR?

**Comment by td-adc** [comment]
Any updates/progress on this?

File: manifests/config.pp [hunk 2]
```diff
@@ -164,7 +164,7 @@
     # }
 
     # Generate Elasticsearch config
-    $data = merge(
+    $data = stdlib::merge(
```
Comment (h-haaks) [inline manifests/config.pp:167]: I would really like to release a new version that allow 9.x before we start requiring 9.x
This makes it a bit easier to upgrade without breaking dependencies.

**Comment by h-haaks** [comment]
@ggabijaa could you please rebase against voxpupuli:master so that it is ready for merge once the current changes on master is released?

**Comment by h-haaks** [comment]
Rename PR and squash commits to something like 'Use namespaced stdlib functions'?

File: manifests/config.pp [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
   $init_defaults = {
     'MAX_OPEN_FILES' => '65535',
-  }.merge($elasticsearch::init_defaults)
+  }.stdlib::merge($elasticsearch::init_defaults)
```
Comment (bastelfreak) [inline manifests/config.pp:24]: you can add the hashes with `+`. then you don't need merge and don't need stdlib.

Comment (h-haaks) [reply manifests/config.pp:24]: #1217 and just close this PR?

**Comment by h-haaks** [comment]
@ggabijaa please have a look at #1217 
We could either merge that or you need to remove the stdlib dependency from metatdata.yaml and clean up your commits.

**Pull request closed without merging**